spyderlib/plugins/shortcuts.py

```diff
@@ -289,12 +255,20 @@ def nonedit_keyrelease(self, e):
                 self.button_ok.setFocus()
 
     def keyReleaseEvent(self, e):
-        """Qt override"""
+        """Qt override."""
         self.npressed -= 1
         if self.npressed <= 0:
-            if len(self.keys) == 1 and (list(self.keys)[0] == Qt.Key_Tab):
+            key = e.key()
+
+            if len(self.keys) == 1 and key == Qt.Key_Tab:
                 self.toggle_state()
 
+            if len(self.keys) == 1 and key == Qt.Key_Escape:
+                self.close()

```

Comment:
Instead of closing the dialog, my suggestion for `Esc` is to do `toggle_state` (as `Tab`) **and** also reset the shortcut present in the new shortcut QLineEdit